### PR TITLE
MAINT: Remove branch in __array__ where if and else were the same

### PR DESCRIPTION
**Pull request by eric-wieser**

### CODE REVIEW

File: numpy/core/src/multiarray/methods.c [hunk 1]
```diff
@@ -973,20 +973,18 @@ array_getarray(PyArrayObject *self, PyObject *args)
...
-        }
 
         Py_INCREF(PyArray_DESCR(self));
-        new = (PyArrayObject *)PyArray_NewFromDescr(subtype,
-                                   PyArray_DESCR(self),
-                                   PyArray_NDIM(self),
-                                   PyArray_DIMS(self),
-                                   PyArray_STRIDES(self),
-                                   PyArray_DATA(self),
-                                   PyArray_FLAGS(self), NULL);
```
Comment (eric-wieser) [inline numpy/core/src/multiarray/methods.c]: There's an awful lot of cases of _"we column-aligned this but then needed to add `(PyArrayObject *)`"_ - may as well fix them as we find them.

**Comment by eric-wieser** [comment]
This was introduced by 6fd31c330e6572963a03b71bcd479dbedd153ac8, which removed `bigndarray` without fully cleaning up

File: numpy/core/src/multiarray/methods.c (hunk 1 above)
Comment (charris) [inline numpy/core/src/multiarray/methods.c]: Weird redundant code, and most of the function is really, really old. One wonders what the intent was? Anyway, the change certainly retains the old behavior.

**Comment by charris** [comment]
Thanks Eric.

File: numpy/core/src/multiarray/methods.c (hunk 1 above)
Comment (eric-wieser) [inline numpy/core/src/multiarray/methods.c]: The intent was to choose between `subdtype = ndarray` and `subdtype = bigndarray`, but then they became the same thing.

**Pull request merged**